File: csharp/1557.minimum-number-of-vertices-to-reach-all-nodes.cs
/*
 * @lc app=leetcode id=1557 lang=csharp
 *
 * [1557] Minimum Number of Vertices to Reach All Nodes
 */

using System.Collections.Generic;
using System.Linq;

// @lc code=start
public class Solution {
    public IList<int> FindSmallestSetOfVertices(int n, IList<IList<int>> edges) {
        var excluded = edges.Select(e => e[1]).ToHashSet();
        return Enumerable
            .Range(0, n)
            .Where(x => !excluded.Contains(x))
            .ToList();
    }
}
// @lc code=end

File: java/1557.minimum-number-of-vertices-to-reach-all-nodes.java
/*
 * @lc app=leetcode id=1557 lang=java
 *
 * [1557] Minimum Number of Vertices to Reach All Nodes
 */

import java.util.List;
import java.util.stream.IntStream;

// @lc code=start
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        var exclude = edges.stream().map(edge -> edge.get(1)).collect(java.util.stream.Collectors.toUnmodifiableSet());
        return IntStream.range(0, n).boxed().filter(x -> !exclude.contains(x)).toList();
    }
}
// @lc code=end

